Library/Formula/node.rb

```diff
@@ -55,42 +55,52 @@ def install
     system "./configure", *args
     system "make", "install"
 
-    resource("npm").stage libexec/"npm" if build.with? "npm"
-  end
+    if build.with? "npm"
+      resource("npm").stage buildpath/"npm_install"
 
-  def post_install
-    return if build.without? "npm"
+      # make sure npm can find node
+      ENV["PATH"] = "#{Formula["node"].prefix}/bin:#{ENV["PATH"]}"
 
-    (libexec/"npm").cd { system "make", "uninstall" }
-    Pathname.glob(HOMEBREW_PREFIX/"share/man/*") do |man|
-      next unless man.directory?
-      man.children.each do |file|
-        next unless file.symlink?
-        file.unlink if file.readlink.to_s.include? "/node_modules/npm/man/"
+      # set log level temporarily for npm's `make install`
+      ENV["NPM_CONFIG_LOGLEVEL"] = "verbose"
+
+      cd buildpath/"npm_install" do
+        system "./configure", "--prefix=#{libexec}/npm"
+        system "make", "install"
       end
+
+      rm_rf libexec/"npm/lib/node_modules/npm/share"
+
+      if build.with? "completion"
+        bash_completion.install \
+          buildpath/"npm_install/lib/utils/completion.sh" => "npm"
+      end
+
+      # Remove the now-redundant directory.
+      rm_rf buildpath/"npm_install"
     end
+  end
+
+  def post_install
+    return if build.without? "npm"
 
     node_modules = HOMEBREW_PREFIX/"lib/node_modules"
     node_modules.mkpath
-    cp_r libexec/"npm", node_modules
+    npm_exec = node_modules/"npm/bin/npm-cli.js"
+    # Kill npm but preserve all other modules across node updates/upgrades.
+    rm_rf node_modules/"npm"
```

Comment:
For the previous version, theoretically, because ` npm ` installs the manpages in ` install_prefix/share/man `, and we were installing ` npm ` directly into ` /usr/local ` before, so I guess it hooked into the directory.

For this install, I'm not quite sure. Whereas the previous version had a decent excuse, we're actively installing this one into ` libexec/npm ` so it shouldn't be able to break out of ` libexec ` so easily. If anything, if it is determined to link the manpages itself, it should be trying to link them to ` libexec/share/man ` but it doesn't.